### PR TITLE
Update to MAPL v2.12.3 and gftl-shared v1.5.0

### LINKED ISSUE #182: Update MAPL to v2.12.3
MAPL v2.11.0 does not build on M1 Mac, complaining about `Unknown processor`. This should be fixed with a newer version of MAPL. A request for MAPL v2.12.3 was made, so let's do it here too.

https://github.com/NOAA-EMC/hpc-stack/issues/460

### PR DESCRIPTION
**Pull request by kgerheiser**
Build worked for me using apple-clang on macOS. Can you try this out @climbfuji?

Fix #182

### CODE REVIEW

**Comment by climbfuji** [comment]
Thanks, yes, will do after the current PRs are merged.

**Comment by kgerheiser** [comment]
I think we still need that patch. The M1 fix is for native aarch64, and not Rosetta.

**Comment by kgerheiser** [comment]
@climbfuji could you try again? I updated ESMA_CMake and cleanup up the MAPL build with https://github.com/NOAA-EMC/spack/pull/91

File: configs/common/packages.yaml [hunk 1]
```diff
@@ -116,7 +116,7 @@
     grib-util:
       version: [1.2.3]
     gftl-shared:
-      version: [1.3.3]
+      version: [1.3.6]
```
Comment (climbfuji) [inline configs/common/packages.yaml]: Should this be 1.5.0? That's what the concretizer picked on my mac.

**Comment by climbfuji** [comment]
@kgerheiser The spack PR was merged, please revert `.gitmodules` and update the submodule pointer for `spack`. Also, wondering about which gftl-shared version to use in `config/common/packages.yaml`.

File: configs/common/packages.yaml (hunk 1 above)
Comment (kgerheiser) [reply configs/common/packages.yaml]: It picked 1.3.6 for me. I have no reason to prefer 1.3.6, so I'm going to change it to 1.5.0 (the latest version) and if it works I'll go with it.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**